File: src/domain/repositories/ICourseRepository.ts
import { Course } from '../entities/Course';
import { Skill } from '../entities/Skill';
import { Question } from '../entities/Question';

export interface ICourseRepository {
  getAllCourses(): Promise<Course[]>;
  getCourseById(courseId: string): Promise<Course | null>;
  getSkillsByCourseId(courseId: string): Promise<Skill[]>;
  getPlacementQuestions(courseId: string): Promise<Question[]>;
}

File: mobile/lib/domain/repositories/i_course_repository.dart
import '../entities/course.dart';
import '../entities/skill.dart';
import '../entities/question.dart';

abstract class ICourseRepository {
  Future<List<Course>> getAllCourses();
  Future<Course?> getCourseById(String courseId);
  Future<List<Skill>> getSkillsByCourseId(String courseId);
  Future<List<Question>> getPlacementQuestions(String courseId);
}
